### PR TITLE
Adding UT for port validate

### PR DESCRIPTION
**Pull request by longkb**
Signed-off-by: Kim Bao Long <longkb@vn.fujitsu.com>

### CODE REVIEW

File: pkg/internal/apis/config/validate_test.go [hunk 1]
```diff
@@ -243,3 +244,48 @@ func TestNodeValidate(t *testing.T) {
...
+
+	for _, tc := range cases {
+		tc := tc //capture loop variable
+		t.Run(tc.TestName, func(t *testing.T) {
+			err := validatePort(tc.Port)
+			// the error can be:
+			// - nil, in which case we should expect no errors or fail
+			if err == nil && tc.ExpectErrors != nil {
+				t.Error("received no errors but expected errors for case")
+			}else if !reflect.DeepEqual(err, tc.ExpectErrors) {
```
Comment (BenTheElder) [inline pkg/internal/apis/config/validate_test.go]: nit: should be `gofmt`ed (which would be `} else if ...`)

Comment (BenTheElder) [reply pkg/internal/apis/config/validate_test.go]: `hack/update/gofmt.sh` should fix this

Comment (longkb) [reply pkg/internal/apis/config/validate_test.go]: Done. Thanks :)

**Comment by aojea** [comment]
the test is panicking, @łongkb I think that we can simplify the test with this approach and avoid using reflect:

```
func TestPortValidate(t *testing.T) {
	cases := []struct {
		TestName    string
		Port        int32
		ExpectError string
	}{
		{
			TestName:    "-1 port",
			Port:        -1,
			ExpectError: "",
		},
		{
			TestName:    "valid port",
			Port:        10,
			ExpectError: "",
		},
		{
			TestName:    "negative port",
			Port:        -2,
			ExpectError: "invalid port number: -2",
		},
		{
			TestName:    "extra port",
			Port:        65536,
			ExpectError: "invalid port number: 65536",
		},
	}

	for _, tc := range cases {
		tc := tc //capture loop variable
		t.Run(tc.TestName, func(t *testing.T) {
			err := validatePort(tc.Port)
			// the error can be:
			// - nil, in which case we should expect no errors or fail
			if err == nil && len(tc.ExpectError) > 0 {
				t.Errorf("Test failed, unexpected error: %v %s", err, tc.ExpectError)
			}

			if err != nil && err.Error() != tc.ExpectError {
				t.Errorf("Test failed, error: %s expected error: %s", err, tc.ExpectError)
			}
		})
	}
}
```

**Comment by longkb** [comment]
> 
> 
> the test is panicking, @longkb I think that we can simplify the test with this approach and avoid using reflect:
> 
> ```go
> func TestPortValidate(t *testing.T) {
> 	cases := []struct {
> 		TestName    string
> 		Port        int32
> 		ExpectError string
> 	}{
> 		{
> 			TestName:    "-1 port",
> 			Port:        -1,
> 			ExpectError: "",
> 		},
> 		{
> 			TestName:    "valid port",
> 			Port:        10,
> 			ExpectError: "",
> 		},
> 		{
> 			TestName:    "negative port",
> 			Port:        -2,
> 			ExpectError: "invalid port number: -2",
> 		},
> 		{
> 			TestName:    "extra port",
> 			Port:        65536,
> 			ExpectError: "invalid port number: 65536",
> 		},
> 	}
> 
> 	for _, tc := range cases {
> 		tc := tc //capture loop variable
> 		t.Run(tc.TestName, func(t *testing.T) {
> 			err := validatePort(tc.Port)
> 			// the error can be:
> 			// - nil, in which case we should expect no errors or fail
> 			if err == nil && len(tc.ExpectError) > 0 {
> 				t.Errorf("Test failed, unexpected error: %s", tc.ExpectError)
> 			}
> 
> 			if err != nil && err.Error() != tc.ExpectError {
> 				t.Errorf("Test failed, error: %s expected error: %s", err, tc.ExpectError)
> 			}
> 		})
> 	}
> }
> ```

Thanks @aojea . It passed the UT with your great patch :)

**Comment by aojea** [comment]
/retest
/lgtm
/assign @BenTheElder 
Thanks

File: pkg/internal/apis/config/validate_test.go [hunk 2]
```diff
@@ -243,3 +243,49 @@ func TestNodeValidate(t *testing.T) {
...
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc //capture loop variable
+		t.Run(tc.TestName, func(t *testing.T) {
+			t.Parallel()
+			err := validatePort(tc.Port)
+			// the error can be:
+			// - nil, in which case we should expect no errors or fail
```
Comment (BenTheElder) [inline pkg/internal/apis/config/validate_test.go]: nit: this comment seems to cover the cases on the surface, but doesn't actually list them ...?

**Comment by BenTheElder** [comment]
/hold
holding for a moment to see if we can handle the one comment nit :-)

**Pull request merged**